src/features/filter.ts

```diff
@@ -1,10 +1,55 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetQueryAction = { type: 'filter/SET_QUERY', payload: string };
+type ClearQueryAction = { type: 'filter/CLEAR_QUERY' };
+type SetStatusAction = { type: 'filter/SET_STATUS', payload: Status };
+
+const setQuery = (query: string): SetQueryAction => ({
+  type: 'filter/SET_QUERY',
+  payload: query,
+});
+
+const clearQuery = (): ClearQueryAction => ({ type: 'filter/CLEAR_QUERY' });
+
+const setStatus = (status: Status): SetStatusAction => ({
+  type: 'filter/SET_STATUS',
+  payload: status,
+});
+
+export const actions = { setQuery, clearQuery, setStatus };
+
+type Action = SetQueryAction | ClearQueryAction | SetStatusAction;
+
+type State = {
+  query: string,
+  status: Status,
```

Comment:
Consider using a more descriptive type name for 'Status' like 'TodoStatus' to make the code more readable 📖